### ISSUE TITLE
GestureRecognizer Action

### ISSUE THREAD

**Issue by cesarmtz93-wit** [ZacharyKhan/ZKCarousel#30]
By default it has the tap gesture that goes to the next image (slide) but is there a way in where i can change that action?

**Comment by cesarmtz93-wit** [comment]
i guess it's pretty simpl, you need to make it public: 

```swift
    public lazy var tapGesture : UITapGestureRecognizer = {
        let tap = UITapGestureRecognizer(target: self, action: #selector(tapGestureHandler(tap:)))
        return tap
    }()
```
 so that we can change the target like such:
```swift
let carousel = ZKCarousel()
carousel.tapGesture.addTarget(<#T##target: Any##Any#>, action: <#T##Selector#>)
```

**Comment by ZacharyKhan** [comment]
Thank you for the suggestion. I will make the selector public and allow users to use their own function for the tap action. I'll reply here once I have pushed the updates.

**Comment by cesarmtz93-wit** [comment]
do you have an estimated date of this release?

**Comment by ZacharyKhan** [comment]
>do you have an estimated date of this release? 

@cesarmtz93-wit I can have it done by tonight, I apologize I’ve had a very busy week. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by cesarmtz93-wit** [comment]
Awesome, yeah any time this well or next week is fine, I was just wondering! 

Thanks man!

**Comment by ZacharyKhan** [comment]
>Awesome, yeah any time this week or next week is fine, I was just wondering!  ...

@cesarmtz93-wit not a problem, it should only take about 20-30 minutes or so. I can take a stab this evening. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by ZacharyKhan** [comment]
@cesarmtz93-wit I've released a new version of ZKCarousel (1.0.2) which includes the changes you requested.

1. I have added a function `disableTap()` which you should call before adding your own custom gesture recognizer. This function simple removes the default recognizer from the UICollectionView. 

2. I have made the UICollectionView within ZKCarousel public, which will allow you to add your own gesture recognizer to the UICollectionView and manipulate the UICollectionView as you wish. **Just be sure to call `disableTap()` before adding your own recognizer to avoid multiple functions being called** 

If you have any other questions or concerns regarding this issue, feel free to continue the conversation here and I'd be happy to reopen the issue if necessary. Otherwise, please open a new issue for anything else. Best of luck!

**Comment by cesarmtz93-wit** [comment]
Awesome man! thanks for the help!
